src/services/users.js

```diff
@@ -0,0 +1,195 @@
+const { User } = require('../models');
+const { UserRoles } = require('../models');
+const { UserStatuses } = require('../models');
+const bcrypt = require('bcryptjs');
+
+class UsersService {
+  async updateUserPassword (req, res) {
+    try {
+      const { oldPassword, newPassword } = req.body;
+      const user = await User.findOne({ _id: req.params.id }).exec();
+
+      return bcrypt.compare(
+        oldPassword,
+        user.password,
+        async function (err, isMatch) {
+          if (err) throw err;
+          if (!isMatch) return res.boom.badRequest('wrong old password');
+          if (oldPassword !== newPassword) {
+            user.password = newPassword;
+            const salt = await bcrypt.genSalt(10);
+            user.password = await bcrypt.hash(user.password, salt);
+            await user.save();
+            return user ? res.json(true) : res.boom.notFound();
+          } else {
+            res.boom.badRequest('new password cannot be the same as old');
+          }
+        }
+      );
+    } catch (err) {
+      res.boom.notFound();
+    }
+  }
+
+  async getStatusActive () {
+    const statusActive = await UserStatuses.findOne({ name: 'Active' });
+    return statusActive;
+  }
+
+  async getRoleUser () {
+    const roleUser = await UserRoles.findOne({ name: 'User' });
+    return roleUser;
+  }
+
+  async createBasicUser (req, res) {
+    const statusActive = await this.getStatusActive();
+    const newUser = await new User(req.body);
+    newUser.status = statusActive._id;
+    const salt = await bcrypt.genSalt(10);
+    newUser.password = await bcrypt.hash(newUser.password, salt);
+    return newUser;
+  }
+
+  async checkUserExists (req, res) {
+    const userExist = await User.findOne({ email: req.body.email });
+    return userExist;
+  }
+
+  async createUser (req, res) {
+    try {
+      const roleUser = await this.getRoleUser();
+      const newUser = await this.createBasicUser(req);
+      newUser.role = roleUser._id;
+      const userExist = await this.checkUserExists(req);
```

Comment:
Check if user exists at start. There is no point to perform 2 calls to db (getStatus and getRole), if user already exists

Reply:
fixed